common/workunit/workunit.cpp

```diff
@@ -2383,6 +2383,40 @@ void CWorkUnitFactory::clearAborting(const char *wuid)
     }
 }
 
+bool CWorkUnitFactory::checkWorkUnitSession(const char *wuid, WUState &state, SessionId agent)
```

Comment:
checkWorkUnitSession returning true =  job has terminated abnormally.

I think a different method name would capture the meaning better.
Maybe, hasSessionTerminated()